### PR TITLE
Fixed CreateContainerIfNotExistsAsync validation to limit to partitionKeyPath only

### LINKED ISSUE #623: CreateContainerIfNotExistsAsync throws ArgumentException incorrectly stating the PKs don't match
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
After upgrading to latest SDK, CreateContainerIfNotExistsAsync throws an ArgumentException stating PK "/x" in Container "foo" does not match PK "/x". The actual paths in the error message are identical though.

**Environment summary**
SDK Version: 3.1.0
OS Version (e.g. Windows, Linux, MacOSX) Windows

### PR DESCRIPTION
**Pull request by simplynaveen20**
# Pull Request Template

## Description

If user had created Container from SDK other than V3 with explicitly specifying 
IsSystemKey = true or false and then trying to recreate with V3 api CreateContainerIfNotExistsAsync  , sdk is throwing error which is should not.
This PR with address it

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues

closes #623

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs [hunk 1]
```diff
@@ -186,9 +186,7 @@ internal DatabaseCore()
             if (response.StatusCode != HttpStatusCode.NotFound)
             {
                 ContainerResponse retrivedContainerResponse = await this.ClientContext.ResponseFactory.CreateContainerResponseAsync(container, Task.FromResult(response));
-                if (!PartitionKeyDefinition.AreEquivalent(
-                        retrivedContainerResponse.Resource.PartitionKey,
-                        containerProperties.PartitionKey))
+                if (!retrivedContainerResponse.Resource.PartitionKeyPath.Equals(containerProperties.PartitionKeyPath))
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs]: Null check partitionKeyPath

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs]: image

**Comment by simplynaveen20** [comment]
> Please add entry into ChangeLog.md

This issue is not major scenario , and not reported by multiple users , also it was introduce in 3.1.0 and not in 3.0.0 so do we want to call it out ?

**Comment by j82w** [comment]
> not reported by multiple users , also it was introduce in 3.1.0 and not in 3.0.0 so do we want to call it out ?
I believe we should still add it to the changelog. We had one customer already hit the issue. The point of the changelog is for customer to see the issues fixed so they know to upgrade without creating a new issue.

File: changelog.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - [#612](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/612) Bug fix for ReadFeed with partition-key
 - [#614](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/614) Fixed SpatialPath serialization and compatibility with older index versions
+- [#629](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/629) Bug fix for CreateContainerIfNotExistsAsync api
```
Comment (kirankumarkolli) [inline changelog.md]: Its very generic, please update with more context. 
How about "CreateContainer.... limiting PK validation to path only

**Comment by kirankumarkolli** [comment]
> > not reported by multiple users , also it was introduce in 3.1.0 and not in 3.0.0 so do we want to call it out ?
> 
> I believe we should still add it to the changelog. We had one customer already hit the issue. The point of the changelog is for customer to see the issues fixed so they know to upgrade without creating a new issue.

Idea is share more context for developers to make deployment decision. 
Less or ambiguous details will hurt new version adoptions.

**Pull request merged**